internal/commands/create_test.go

```diff
@@ -104,6 +106,64 @@ import data.lib.foo`
 	}
 }
 
+func TestLoadPolicyFiles_MissingViolationRule(t *testing.T) {
+	policyMissingViolationRule := `package test
+default a = true`
+
+	file, err := ioutil.TempFile("", "policy")
```

Comment:
opinion: We should make a strong effort to not need temporary files in our tests.

One of the nice things about Go is that because the tests are (often) in the same package as the functions under test, we have access to the "private" methods. By instead testing crafting our own `map[string]string` ahead of time and passing that in, we can completely side step the need for a temporary file.

Yes, technically code coverage goes down, but more often that not the extra coverage doesn't really test anything (we should trust Go can open a file, read it, put it somewhere), and it greatly complicates our test suite.

Instead, I would try and make the loadLibrary/loadPolicy functions take in the input that we get from `readFilesContents`, that is, `map[string]string`. There really isn't much to test to _get_ that data, where the testing comes in is how we analyze that data (whats a policy, whats a library).

Then we can remove the temporary file stuff completely.

This is a common pattern that I leverage a lot. A more straight-forward example would be:

```go
func GetFileContents(path string) string {
  data, _ := ioutil.ReadFile(path)

 return getContents(data)
}

func getContents(data string) string {
  return data
}
```

Then we actually test `getContents`, because we shouldn't care much _where_ the contents comes from. We trust that Go can open a file, and then we test the logic that our function is supposed to do when given a string. Regardless if that string comes from a file, user input, etc.

Reply:
That makes sense. I believe I've addressed this one, let me know if it looks OK.